### ISSUE TITLE
using Optional in TypedDict provides not consistent behaviour with PEP 589

### ISSUE THREAD

**Issue by p-malecki** [google/pytype#1549]
#### Example Code
```python 
from typing_extensions import TypedDict
from typing import Optional


class A(TypedDict):
    x: Optional[int]

class B(TypedDict):
    x: int

def f(a: A) -> None:
    a['x'] = None

b: B = {'x': 0}
f(b)  # Type check error: 'B' not compatible with 'A'
b['x'] + 1  # Runtime error: None + 1
```

#### Expected Behavior
error:  incompatible type argument in function f
#### Actual Behavior
```
Success: no errors found
```
Example works as expected in mypy.

**Comment by rchen152** [comment]
Agreed, pytype should flag this. We (the pytype devs) discussed this briefly, and we probably want to match mutable and immutable container types differently.

**Comment by h-joo** [comment]
Hi, thank you for your report. Google is shifting its effort to a different approach for type checking apart from pytype, and we're not planning to put in any effort in the near future in pytype, so we'll close this issue. Please see the announcement [here](https://github.com/google/pytype/blob/main/README.md).
